Sparkle/SUFileManager.m

```diff
@@ -150,18 +140,12 @@ - (int)_removeXAttr:(NSString *)name fromFile:(NSString *)file options:(int)opti
         return -1;
     }
     
-    const char *attr = [name cStringUsingEncoding:NSASCIIStringEncoding];
-    if (attr == NULL) {
-        errno = 0;
-        return -1;
-    }
-    
     return removexattr(path, attr, options);
 }
 
 #define XATTR_UTILITY_PATH "/usr/bin/xattr"
 // Recursively remove an xattr at a specified root URL with authentication
-- (BOOL)_removeXAttrWithAuthentication:(NSString *)name fromRootURL:(NSURL *)rootURL error:(NSError *__autoreleasing *)error
+- (BOOL)_removeXAttrWithAuthentication:(char *__nonnull)xattrName fromRootURL:(NSURL *)rootURL error:(NSError *__autoreleasing *)error
 {
```

Comment:
Yeah. https://developer.apple.com/swift/blog/?id=25

__Nonnull and __Nullable are more modern to use but __nonnull and __nullable still exist for compatibility I suppose.